### PR TITLE
Create GitHub actions

### PR DESCRIPTION
**Pull request by lockieRichter**
Creates a GitHub actions file that runs the tests and builds iOS, Android and Web projects.

### CODE REVIEW

**Comment by shockbytes** [comment]
I think this link might be useful: https://stackoverflow.com/a/70617283/3111388

Let me know if you can add secrets to the repository, otherwise, I can add the `google-services.json` file.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        run: flutter build apk
+        
+      - name: Build Android Appbundle
+        run: flutter build appbundle
+        
+      - name: Build iOS
+        run: flutter build ios --release --no-codesign
+        
+      - name: Build Web
+        run: flutter build web
```
Comment (shockbytes) [inline .github/workflows/main.yml]: For your information: We also have a codemagic.io setup that is responsible for creating the test builds for Android and iOS. 

I think it's good to keep these builds here as well, so we know that it also works for every PR.

Only remark: I don't think we need both, APK and Android App Bundle just for build verification. I think we can drop the APK build.

Comment (lockieRichter) [reply .github/workflows/main.yml]: Yeah agreed, I think it's super useful for the PR's to have build and test information. Will remove APK build.

Comment (lockieRichter) [reply .github/workflows/main.yml]: I'll have a go at adding the secrets to the repo and if I can't will get you to do it.

**Comment by lockieRichter** [comment]
@shockbytes I can't add secrets, but I've updated the script to look for a `google_services_json` secret in the runner, can you add in please?

**Comment by lockieRichter** [comment]
Also would be really cool to add our code coverage icon to our repo using Codecov -> https://damienaicheh.github.io/flutter/github/actions/2021/05/06/flutter-tests-github-actions-codecov-en.html

**Comment by shockbytes** [comment]
@lockieRichter added the secret but we are facing now another error while creating the file.

**Comment by lockieRichter** [comment]
@shockbytes need you to add a secret for GOOGLE_SERVICES_INFO_PLIST, the same as the json file please!

**Comment by shockbytes** [comment]
Added `GOOGLE_SERVICES_INFO_PLIST` secret too but the build was already working, no?

**Comment by lockieRichter** [comment]
@shockbytes yeah it must have worked with an empty file, good to have the secret there though if we ever want to build proper files here. Ready for final review now.

**Review by shockbytes — APPROVED** [review]

**Pull request merged**